### ISSUE TITLE
TagCloseButton missing aria-label or title makes it inaccessible to screen readers

### ISSUE THREAD

**Issue by AnanaMJ** [chakra-ui/chakra-ui#3569]
# 🐛 Bug report

When using a Tag with close button, there is no title or aria-label defined, so VoiceOver only reads 'button' instead of 'close, button'.

## 💥 Steps to reproduce

1. Go to https://codesandbox.io/s/thirsty-cookies-tt8xs?file=/src/index.tsx
2. Inspect element from the close button
3. See that there are no accessibility properties.

## 💻 Link to reproduction

https://codesandbox.io/s/thirsty-cookies-tt8xs?file=/src/index.tsx

## 🧐 Expected behavior

VoiceOver should read 'close, button'.

## 🧭 Possible Solution

Add `title='close'` or `aria-label='close'` to `TagCloseButton` component.

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Chakra UI        |      1.3.3      |
| Browser          |     Firefox       |
| Operating System |       MacOS     |

## 📝 Additional information

- I used version 1.3.3 of Chakra UI because in 1.3.4, Tags are broken. There's something about getting undefined instead of string.
